File: 3151/main.go
package main

import (
	"fmt"
	"os"
)

func main() {
	fmt.Println(isArraySpecial([]int{4,1,2,3}))
	fmt.Fprintln(os.Stderr, " + ")
}

func isArraySpecial(nums []int) bool {
	ans := true 
	if len(nums) == 1 {
		return true
	}
	for i := 0; i < len(nums)-1; i++ {
		if (nums[i] + nums[i+1]) % 2 == 0 {
			return false
		}
	}
	return ans
}

File: 3151/main.rs

struct Solution{}

impl Solution {
    pub fn is_array_special(nums: Vec<i32>) -> bool {
        if nums.len() == 1 {
            return true
        }

        for i in 0..nums.len()-1 {
            if (nums[i+1] + nums[i]) % 2 == 0 {
                return false
            }
        }
        true
    }
}

fn main() {
    println!("{}", Solution::is_array_special(vec![4,1,2,3]))
}